File: mobile/lib/models/food_truck.dart
class FoodTruck {
  final int id;
  final String name;
  final String foodType;
  final String locationDescription;
  final double latitude;
  final double longitude;
  final String? menuInfo;
  final String? news;
  final String reportedBy;
  final DateTime lastReportedAt;
  final String lastReportedHuman;

  FoodTruck({
    required this.id,
    required this.name,
    required this.foodType,
    required this.locationDescription,
    required this.latitude,
    required this.longitude,
    this.menuInfo,
    this.news,
    required this.reportedBy,
    required this.lastReportedAt,
    required this.lastReportedHuman,
  });

  factory FoodTruck.fromJson(Map<String, dynamic> json) {
    return FoodTruck(
      id: json['id'],
      name: json['name'],
      foodType: json['food_type'],
      locationDescription: json['location_description'],
      latitude: json['latitude'].toDouble(),
      longitude: json['longitude'].toDouble(),
      menuInfo: json['menu_info'],
      news: json['news'],
      reportedBy: json['reported_by'],
      lastReportedAt: DateTime.parse(json['last_reported_at']),
      lastReportedHuman: json['last_reported_human'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'food_type': foodType,
      'location_description': locationDescription,
      'latitude': latitude,
      'longitude': longitude,
      'menu_info': menuInfo,
      'news': news,
      'reported_by': reportedBy,
      'last_reported_at': lastReportedAt.toIso8601String(),
      'last_reported_human': lastReportedHuman,
    };
  }
}

File: backend/app/Models/FoodTruck.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Carbon\Carbon;

class FoodTruck extends Model
{
    use HasFactory;

    protected $fillable = [
        'name',
        'food_type',
        'location_description',
        'latitude',
        'longitude',
        'menu_info',
        'news',
        'reported_by',
        'last_reported_at',
        'is_active'
    ];

    protected $casts = [
        'latitude' => 'decimal:8',
        'longitude' => 'decimal:8',
        'last_reported_at' => 'datetime',
        'is_active' => 'boolean'
    ];

    // Automatically set last_reported_at when creating
    protected static function boot()
    {
        parent::boot();
        
        static::creating(function ($foodTruck) {
            if (!$foodTruck->last_reported_at) {
                $foodTruck->last_reported_at = now();
            }
        });
    }

    // Scope for active trucks only
    public function scopeActive($query)
    {
        return $query->where('is_active', true);
    }

    /**
     * Get the location reports for this food truck
     */
    public function locationReports(): HasMany
    {
        return $this->hasMany(LocationReport::class);
    }

    /**
     * Get pending location reports for this food truck
     */
    public function pendingLocationReports(): HasMany
    {
        return $this->hasMany(LocationReport::class)->where('status', 'pending');
    }

    // Format for API response
    public function toApiArray()
    {
        $lastReportedAt = $this->last_reported_at ? Carbon::parse($this->last_reported_at) : null;
        
        return [
            'id' => $this->id,
            'name' => $this->name,
            'food_type' => $this->food_type,
            'location_description' => $this->location_description,
            'latitude' => (float) $this->latitude,
            'longitude' => (float) $this->longitude,
            'menu_info' => $this->menu_info,
            'news' => $this->news,
            'reported_by' => $this->reported_by,
            'last_reported_at' => $lastReportedAt ? $lastReportedAt->format('c') : null,
            'last_reported_human' => $lastReportedAt ? $lastReportedAt->diffForHumans() : null,
        ];
    }
}
